plutus-core/plutus-core/src/PlutusCore/Builtin/TypeScheme.hs

```diff
@@ -34,15 +34,35 @@ import GHC.TypeLits
 
 infixr 9 `TypeSchemeArrow`
 
+{- Note [MakeKnown in TypeSchemeArrow]
+There's a @MakeKnown val arg@ constrained packed in the 'TypeSchemeArrow' constructor. It's not
+supposed to be there, but unfortunately, in the @Generators@ tests we use 'TypeScheme' for
+generation of arbitrary arguments of builtins and that requires 'makeKnown', which makes us have
+the 'MakeKnown' in 'TypeSchemeArrow'.
+
+The solution is to fix the @Generators@ tests. Explicitly constraining @args@ outside of
+'TypeScheme' sounds like a promising strategy. Maybe we could just delete those tests altogether.
+
+However it's also worth considering untangling 'RuntimeScheme' from 'TypeScheme' and generating the
+two in parallel, so that we only need to optimize the former. Then we will be able to afford having
+any kind of nonsense in 'TypeScheme'. Another reason for that would be the fact that Core output
```

Comment:
Yeah, this seems reasonable. It doesn't seem crazy that a value representing a type signature for a function should allow you to both lift and unlift the argument types. It's just that we don't need that at runtime, and that's what we have `RuntimeScheme` for.

Reply:
Not sure what you're saying. That what we have right now is already fine, given that `MakeKnown` only appears for arguments in `TypeScheme` and not `RuntimeScheme`?

Reply:
Yes, which I think is also what you were saying in the comment: that it seems *conceptually* fine, so long as it doesn't leak into the implementation of the `RuntimeScheme`s.